### ISSUE TITLE
Not including https throw error?

### ISSUE THREAD

**Issue by akgold** [rstudio/pins#57]
This breaks because server is `https`. Any way to handle more gracefully? Most rsconnect instances are going to be `https`, so possible to try automatically? Or at least to warn on that? Perhaps on connect step?

```
> pins::board_register_rsconnect(server = "colorado.rstudio.com/rsc", key = Sys.getenv("RSCONNECT_API_KEY"))
> pins::board_connect("rsconnect")
> pins::pin(mtcars, "mtcars", board = "rsconnect")
No encoding supplied: defaulting to UTF-8.
Error in board_pin_create.rsconnect(board, store_path, name = name, metadata = metadata,  : 
  Failed to create pin: Operation failed with status 404: 404 page not found
```

Changing URL works fine:
```
> pins::board_register_rsconnect(server = "https://colorado.rstudio.com/rsc", key = Sys.getenv("RSCONNECT_API_KEY"))
> pins::board_connect("rsconnect")
> pins::pin(mtcars, "mtcars", board = "rsconnect")
# A tibble: 32 x 11
     mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb
   <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
 1  21       6  160    110  3.9   2.62  16.5     0     1     4     4
 2  21       6  160    110  3.9   2.88  17.0     0     1     4     4
 3  22.8     4  108     93  3.85  2.32  18.6     1     1     4     1
 4  21.4     6  258    110  3.08  3.22  19.4     1     0     3     1
 5  18.7     8  360    175  3.15  3.44  17.0     0     0     3     2
 6  18.1     6  225    105  2.76  3.46  20.2     1     0     3     1
 7  14.3     8  360    245  3.21  3.57  15.8     0     0     3     4
 8  24.4     4  147.    62  3.69  3.19  20       1     0     4     2
 9  22.8     4  141.    95  3.92  3.15  22.9     1     0     4     2
10  19.2     6  168.   123  3.92  3.44  18.3     1     0     4     4
# … with 22 more rows
Warning messages:
1: invalid uid value replaced by that for user 'nobody' 
2: invalid gid value replaced by that for user 'nobody'
```

**Comment by javierluraschi** [comment]
Looks fixed in `sparklyr 1.0.4`.

**Comment by JosiahParry** [comment]
Just encountered an interesting related issue using pins `v0.4.0`.

I registered a board as 

```
pins::board_register_rsconnect(server = "connect.rstudioservices.com",
                               account = "josiah",
                               key = Sys.getenv("connect_api"))
```

i was able to pin an object using  `pins::pin(x, name = "x-name", board = "rsconnect")`

I later removed the pin using `pins::pin_remove("x-name")`.

I tried to pin the same object and that failed. The work around was re-registering the board with `https` i unfortunately do not have a reprex to provide.
